pkg/cloud/azure/actuators/cluster/actuator.go

```diff
@@ -63,32 +64,36 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 
 	defer scope.Close()
 
-	networkSvc := network.NewService(scope)
-	resourcesSvc := resources.NewService(scope)
-	certSvc := certificates.NewService(scope)
+	services, err := NewServices(scope)
+	if err != nil {
+		return errors.Errorf("failed to create cluster services: %+v", err)
+	}
 
-	// Store cert material in spec.
-	if err := certSvc.ReconcileCertificates(); err != nil {
-		return errors.Wrapf(err, "failed to reconcile certificates for cluster %q", cluster.Name)
+	err = services.Reconcile()
+	if err != nil {
+		return errors.Errorf("failed to reconcile cluster services: %+v", err)
 	}
 
-	if err := resourcesSvc.ReconcileResourceGroup(); err != nil {
-		return errors.Wrapf(err, "failed to reconcile resource group for cluster %q", cluster.Name)
+	networkSvc, err := network.NewService(scope)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create network service %q", scope.Cluster.Name)
```

Comment:
Let's change this a bit since this line has been touched in the PR :)
```suggestion
		return errors.Wrapf(err, "failed to create network service %s", scope.Cluster.Name)
```